src/main/java/org/elasticsearch/search/highlight/PlainHighlighter.java

```diff
@@ -124,7 +124,7 @@ public HighlightField highlight(HighlighterContext highlighterContext) {
                 }
             }
         } catch (Exception e) {
-            if (e instanceof BytesRefHash.MaxBytesLengthExceededException) {
+            if (e.getCause() instanceof BytesRefHash.MaxBytesLengthExceededException) {
```

Comment:
can we use here `ExceptionHelper.unwrap(e, BytesRefHash.MaxBytesLengthExceededException)` and check if it doesn't return null, we can handle it? I think its safer to use this logic in master as well, its safer?